lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1917,269 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
             return iops_dict[min_iops]
         else:
             raise LisaException(f"Data disk type {disk_type} is unsupported.")
+
+
+def add_system_assign_identity(
+    credential: DefaultAzureCredential,
+    subscription_id: str,
+    resource_group_name: str,
+    vm_name: str,
+    location: str,
+    log: Logger,
+) -> Any:
+    compute_client = ComputeManagementClient(credential, subscription_id)
+    params_identity = {"type": "SystemAssigned"}
+    params_create = {"location": location, "identity": params_identity}
+
+    try:
+        vm_poller = compute_client.virtual_machines.begin_create_or_update(
+            resource_group_name,
+            vm_name,
+            params_create,
+        )
+        vm_result = vm_poller.result()
+        object_id_vm = vm_result.identity.principal_id
+        log.info(object_id_vm)
+
+        if not object_id_vm:
+            raise ValueError("object_id_vm is not set.")
+
+        return object_id_vm
+
+    except Exception as e:
+        log.error(f"Error creating VM: {e}")
+        raise LisaException(str(e))
+
+
+def get_key_vault_management_client(
+    credential: DefaultAzureCredential, subscription_id: str
+) -> KeyVaultManagementClient:
+    return KeyVaultManagementClient(credential, subscription_id)
+
+
+def create_keyvault(
+    credential: DefaultAzureCredential,
+    subscription_id: str,
```

Comment:
use `platform: "AzurePlatform"` then use platform.credential, platform.subscription_id in this method.